packages/jest-snapshot/src/index.ts

```diff
@@ -213,6 +231,48 @@ export const toMatchSnapshot: MatcherFunctionWithContext<
   });
 };
 
+export const toMatchNamedSnapshot: MatcherFunctionWithContext<
+  Context,
+  [snapshotName: string, properties?: object]
+> = function (received, snapshotName, properties?) {
```

Comment:
Ideally all values should go through validation. For example, we should not assume that `snapshotName` will always be `string`. Could you add validation for `snapshotName`, please?

Typings could be altered as in below suggestion to make sure that the values will be validated:

```suggestion
> = function (received: unknown, snapshotName: unknown, properties?: unknown) {
```

Reply:
Added the validation on the matcher function. From my understanding, we have to validate all the arguments because someone might not use typescript, right?

Reply:
Yep, you understand it right.